### ISSUE TITLE
怎样得到saliency map?

### ISSUE THREAD

**Issue by ChunmengLiu1** [NUST-Machine-Intelligence-Laboratory/nsrom#2]
您好，看到文章中使用了saliency map进行背景抽取。请问saliency map是怎样的方式得到的呢？参考哪篇论文呢？可否给出参考的论文title呢？
谢谢您！

**Comment by chentao2016** [comment]
We download the saliency maps from this project: https://github.com/PengtaoJiang/OAA. According to their paper, their saliency maps are generated following the paper of " Deeply supervised salient object detection with short connections."

**Comment by ChunmengLiu1** [comment]
Thanks for your reply. 
There are still some questions.
Firstly,Table 1 is compared to previous state-of-the-art approaches with the VGG backbone. VGG backbone is classification network or segmentation network?
Secondly, Table 2 is compared to previous state-of-the-art approaches with the ResNet backbone.ResNet is a classification network or segmentation network? Without '*' mean the model is pre-trained on Imagenet? If so, could  you provide the pre-trained model on Imagenet for me?
Thirdly, the final result 70.4 is got post-CRF? or 70.4 is got directly from segmentation results without CRF?
I look forward to hearing from you, thank you!
Best wishes to you.

**Comment by chentao2016** [comment]
Both Table 1 and 2 are for the segmentation network. Classification network is for generating pseudo labels.

For imagenet-pretrained weights, you can transform torchvision models to DeepLab(https://github.com/kazuto1011/deeplab-pytorch/issues/67).

The final result 70.4 is got with post-CRF.

**Comment by xinyuaning** [comment]
You are so nice! Thank you very much! 
Also，I find it difficult to find the code of the segmentation network between the VGG backbone(pre-trained by ImageNet) and the ResNet backbone(pretrained by ImageNet). Would you mind providing more information about them? It would be great if you could provide me with the code. (My email address: 2778306060@qq.com, or 1910661@tongji.edu.cn  )

**Comment by ChunmengLiu1** [comment]
Hi！thank you for your advice.
I finished the Res101 coding downloading pre-trained weight from torchvision by setting " pre-trained = True", and also changed the model framework as you mentioned above. Is that what you do?
I also used ImageNet as a VGG16 model for pre-trained weights. I delete "avgpool" and "classifier" and add _ASPP( I don't change the params of the _ASPP ) module, but I got a very weird result, about 22%. I think there are some wrongs, but I can't find them. would you mind share me your experiences with VGG16 training for segmentation? Specifically, how do you structure the network?
This is very important to me. I am waiting for your reply. 
Best wishes to you!

**Comment by chentao2016** [comment]
Sorry for the late reply.

Yes, for the resnet101, I followed the suggestion of dynamic modification.

For the VGG16, I used this code provided in this project (https://github.com/NUST-Machine-Intelligence-Laboratory/ccd/blob/master/model/deeplab_vgg.py).  I also used the initial VGG16 model provided in this project for initialization.

**Comment by ChunmengLiu1** [comment]
Thank you for your response.&nbsp;
I will try it.&nbsp;
Best wishes to you.🌹



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2021年9月18日(周六) 下午5:24
收件人: ***@***.***&gt;;
抄送: ***@***.******@***.***&gt;;
主题: Re: [NUST-Machine-Intelligence-Laboratory/nsrom] 怎样得到saliency map? (#2)




 
Sorry for the late reply.
 
Yes, for the resnet101, I followed the suggestion of dynamic modification.
 
For the VGG16, I used this code provided in this project (https://github.com/NUST-Machine-Intelligence-Laboratory/ccd/blob/master/model/deeplab_vgg.py).  I also used the initial VGG16 model provided in this project for initialization.
 
—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Comment by ChunmengLiu1** [comment]
Hi, I'm sorry to disturb you again. I'm a student of TongJi university in China who is following  your job.
There are still some problems with training vgg backbone.
I used deepklab_vgg.py model as you mentioned above, then I can get a normal results with my own method. However, I think my training strategy is not optimal. Could you share your training strategy with me if you like?( 2778306060@qq.com)
